lib/include/ert/nexus/nexus_plot.hpp

```diff
@@ -0,0 +1,39 @@
+/*
+   Copyright (C) 2017 Statoil ASA, Norway.
+
+   The file 'nexus_plot.hpp' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#ifndef NEXUS_PLOT_H
+#define NEXUS_PLOT_H
+
+#include <cinttypes>
+#include <exception>
+#include <fstream>
+#include <string>
+
+namespace nex {
+
+class bad_header : public std::exception {};
+
+class NexusPlot {
+public:
+    static NexusPlot load( const std::string& );
+    static NexusPlot load( std::istream& );
```

Comment:
So, we did not load in the constructor as this could let users of the API create synthetic nexus states by constructing an empty NexusPlot object and assigning values. The result would also be a constructor doing the full parsing of a nexus file which is somewhat outside the domain of a constructor. I have created a new PR https://github.com/Statoil/libecl/pull/200 that continuous in the direction I was thinking.